producer/src/main/java/com/flipkart/varadhi/produce/providers/TopicProvider.java

```diff
@@ -0,0 +1,73 @@
+package com.flipkart.varadhi.produce.providers;
+
+import com.flipkart.varadhi.common.EntityProvider;
+import com.flipkart.varadhi.entities.VaradhiTopic;
+import com.flipkart.varadhi.entities.auth.ResourceType;
+import com.flipkart.varadhi.spi.db.MetaStore;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Provider for Varadhi topic entities with caching capabilities.
+ * <p>
+ * This class extends the generic {@link EntityProvider} to provide specialized
+ * topic management functionality. It maintains an in-memory cache of topics
+ * that can be preloaded and dynamically updated through entity events.
+ * <p>
+ * The provider is designed for high-performance, thread-safe access to topic
+ * information in a distributed environment. It supports:
+ * <ul>
+ *   <li>Efficient topic lookup by name</li>
+ *   <li>Preloading topics from the metadata store</li>
+ *   <li>Automatic cache updates via entity events</li>
+ * </ul>
+ *
+ * @see EntityProvider
+ * @see VaradhiTopic
+ */
+@Slf4j
+public class TopicProvider extends EntityProvider<VaradhiTopic> {
```

Comment:
as mentioned in previous comment. These classes have less value and we can get away with them if we just use "strategy" pattern / composition.